helpers/etcd.py

```diff
@@ -45,7 +46,7 @@ def put_client_path(self, path, data):
             base64string = base64.encodestring('%s:%s' % (self.authentication["username"], self.authentication["password"])).replace('\n', '')
             request.add_header("Authorization", "Basic %s" % base64string)
         request.get_method = lambda: 'PUT'
-        urllib2.urlopen(request).read()
+        urllib2.urlopen(request, timeout=self.timeout).read()
```

Comment:
It throws a `urllib2.URLError` It gets caught - and spin-locks with sleep until it can contact etcd again. Then it starts the leader race/decision making again and comes to a stable state